x/dao/abci_test.go

```diff
@@ -0,0 +1,568 @@
+		})
+	}
+}
+
+func TestEndBlocker_WithdrawReward(t *testing.T) {
+	validatorReward := sdk.NewInt64Coin(sdk.DefaultBondDenom, 1000000000)
+	expectedDaoFullReward := sdk.NewInt64Coin(sdk.DefaultBondDenom, 1486956434)
+
+	type args struct {
+		vals            map[string]simapp.ValReq
+		treasuryBalance sdk.Coin
+	}
+
+	type wantAssertion struct {
+		vals            map[string]valAssertion
+		treasuryBalance sdk.Coin
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want wantAssertion
+	}{
+		{
+			name: "positive",
+			args: args{
+				vals: map[string]simapp.ValReq{
+					"val1": { // bonded
+						SelfBondCoin: tenBondCoins,
+						Commission:   lowCommission,
+						Reward:       validatorReward,
+					},
+					"val2": { // bonded
+						SelfBondCoin: tenBondCoins,
+						Commission:   lowCommission,
+						Reward:       validatorReward,
+					},
+					"val3": { // won't be bonded
+						SelfBondCoin: nanoBondCoins,
+						Commission:   lowCommission,
+						Reward:       validatorReward,
+					},
+				},
+				treasuryBalance: hundredBondCoins,
+			},
+			want: wantAssertion{
+				vals: map[string]valAssertion{
+					"val1": {
+						bondStatus:     stakingtypes.Bonded,
+						selfBondAmount: tenBondCoins.Amount.ToDec(),
+						daoBondAmount:
+						// initial dao staking
+						hundredBondWithoutStakingPoolRate.QuoInt64(2).
+							// the Reward
+							Add(expectedDaoFullReward.Amount.ToDec().QuoInt64(2).Mul(sdk.OneDec().Sub(types.DefaultStakingTokenPoolRate))).TruncateDec(),
+					},
+					"val2": {
+						bondStatus:     stakingtypes.Bonded,
+						selfBondAmount: tenBondCoins.Amount.ToDec(),
+						daoBondAmount:
+						// initial dao staking
+						hundredBondWithoutStakingPoolRate.QuoInt64(2).
+							// the Reward
+							Add(expectedDaoFullReward.Amount.ToDec().QuoInt64(2).Mul(sdk.OneDec().Sub(types.DefaultStakingTokenPoolRate))).TruncateDec(),
+					},
+					"val3": {
+						bondStatus:     stakingtypes.Unbonded,
+						selfBondAmount: nanoBondCoins.Amount.ToDec(),
+						daoBondAmount:  sdk.ZeroDec(),
+					},
+				},
+				treasuryBalance: sdk.NewCoin(sdk.DefaultBondDenom,
+					sdk.TokensFromConsensusPower(5, sdk.DefaultPowerReduction).
+						Add(expectedDaoFullReward.Amount.ToDec().Mul(types.DefaultStakingTokenPoolRate).RoundInt())),
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			const withdrawRewardPeriod = 6 // the simApp.BeginNextBlock() in assertion will be executed with that block number
+			simApp, _ := createSimAppWithValidatorsAndTreasury(t, tt.args.vals, tt.args.treasuryBalance)
+			simApp.BeginNextBlock()
+			ctx := simApp.NewNextContext()
+			// update dao params to withdraw Reward
+			daoKeeper := simApp.OnomyApp().DaoKeeper
+			daoParams := daoKeeper.GetParams(ctx)
+			daoParams.WithdrawRewardPeriod = withdrawRewardPeriod // withdraw Reward each 10 block
```

Comment:
The comment is removed. 
The loop in the test tests that in case the state of validators remains the same the ReBalance function won't update anything, that's why we shouldn't involve the "withdrawRewardPeriod" here since it might update the state by adding the reward to the DAO pool.